### PR TITLE
ogre2: explicitly request OpenGL 3.3 core profile context.

### PR DESCRIPTION
**Pull request by mahiuchun**
This change raises system requirements as it uses GLX 1.3 API and a GLX extension. Modern systems like Ubuntu bionic and focal satisfies these without issue.

Tested on bionic without GPU (QEMU) and focal with GPU.

# Bug Report

Fixes issue #128 and similar issues.

## Summary
Newer GLX APIs are used to explicitly request core profile GL context that Ogre2 uses. 

Reference:
https://www.khronos.org/opengl/wiki/Tutorial:_OpenGL_3.0_Context_Creation_(GLX)

Reproduce:
Run `ign gazebo shapes.sdf` with the default ogre2 renderer on systems with non-AMD (e.g. Intel) mesa GL drivers or in situations where the hardware GPU is not accessible (e.g. QEMU VM).

The original code has a decent chance to fail due to it attempting to create OpenGL context in compatibility profile and the GL version of context created would be below the required 3.3.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [X] `codecheck` passed (See
  [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See
  [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review
[another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+archived%3Afalse+)
to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]
LGTM, @iche033 to take a second pass.

File: ogre2/src/Ogre2RenderEngine.cc [hunk 1]
```diff
@@ -346,24 +347,51 @@ void Ogre2RenderEngine::CreateContext()
...
       RootWindow(this->dummyDisplay, screenId), 0, 0, 1, 1, 0, 0, 0);
 
-  this->dummyContext = glXCreateContext(x11Display, this->dataPtr->dummyVisual,
-                                        nullptr, 1);
+  PFNGLXCREATECONTEXTATTRIBSARBPROC glXCreateContextAttribsARB = 0;
+  glXCreateContextAttribsARB =
+      (PFNGLXCREATECONTEXTATTRIBSARBPROC)
+      glXGetProcAddress((const GLubyte*)"glXCreateContextAttribsARB");
+
+  if (glXCreateContextAttribsARB) {
```
Comment (iche033) [inline ogre2/src/Ogre2RenderEngine.cc:378]: `{` bracket on a new line

File: ogre2/src/Ogre2RenderEngine.cc [hunk 2]
```diff
@@ -346,24 +347,51 @@ void Ogre2RenderEngine::CreateContext()
...
+      GLX_CONTEXT_MAJOR_VERSION_ARB, 3,
+      GLX_CONTEXT_MINOR_VERSION_ARB, 3,
+      GLX_CONTEXT_FLAGS_ARB        , GLX_CONTEXT_CORE_PROFILE_BIT_ARB,
+      None
+    };
+    this->dummyContext =
+        glXCreateContextAttribsARB(x11Display,
+                                  this->dataPtr->dummyFBConfigs[0], nullptr,
+                                  1, context_attribs);
+  } else {
```
Comment (iche033) [inline ogre2/src/Ogre2RenderEngine.cc:389]: `else` on new line and `{` also on it's own line, ie.e

```
}
else
{
...
}
```

File: ogre2/src/Ogre2RenderEngine.cc [hunk 3]
```diff
@@ -346,24 +347,51 @@ void Ogre2RenderEngine::CreateContext()
...
 
-  this->dummyContext = glXCreateContext(x11Display, this->dataPtr->dummyVisual,
-                                        nullptr, 1);
+  PFNGLXCREATECONTEXTATTRIBSARBPROC glXCreateContextAttribsARB = 0;
+  glXCreateContextAttribsARB =
+      (PFNGLXCREATECONTEXTATTRIBSARBPROC)
+      glXGetProcAddress((const GLubyte*)"glXCreateContextAttribsARB");
+
+  if (glXCreateContextAttribsARB) {
+    int context_attribs[] = {
```
Comment (iche033) [inline ogre2/src/Ogre2RenderEngine.cc:379]: use camelCase, i.e. `contextAttribs[]`

**Review by iche033 — COMMENTED** [review]
looks good to me. Just some very minor coding style comments so that they are consistent with ignition coding style

**Comment by mahiuchun** [comment]
done

**Review by iche033 — APPROVED** [review]

**Pull request merged**